src/blob_info.rs

```diff
@@ -0,0 +1,504 @@
+use std::fmt;
+
+use rlp::{Decodable, DecoderError, Encodable, Rlp, RlpStream};
+
+use super::{
+    common::G1Commitment as DisperserG1Commitment,
+    disperser::{
+        BatchHeader as DisperserBatchHeader, BatchMetadata as DisperserBatchMetadata,
+        BlobHeader as DisperserBlobHeader, BlobInfo as DisperserBlobInfo,
+        BlobQuorumParam as DisperserBlobQuorumParam,
+        BlobVerificationProof as DisperserBlobVerificationProof,
+    },
+};
+
+#[derive(Debug)]
+pub enum ConversionError {
+    NotPresentError,
+}
+
+impl fmt::Display for ConversionError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            ConversionError::NotPresentError => write!(f, "Failed to convert BlobInfo"),
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct G1Commitment {
+    pub x: Vec<u8>,
+    pub y: Vec<u8>,
+}
+
+impl G1Commitment {
+    pub fn to_bytes(&self) -> Vec<u8> {
+        let mut bytes = vec![];
+        bytes.extend(&self.x.len().to_be_bytes());
+        bytes.extend(&self.x);
+        bytes.extend(&self.y.len().to_be_bytes());
+        bytes.extend(&self.y);
+
+        bytes
+    }
+}
+
+impl Decodable for G1Commitment {
+    fn decode(rlp: &Rlp) -> Result<Self, DecoderError> {
+        let x: Vec<u8> = rlp.val_at(0)?; // Decode first element as Vec<u8>
+        let y: Vec<u8> = rlp.val_at(1)?; // Decode second element as Vec<u8>
+
+        Ok(G1Commitment { x, y })
+    }
+}
+
+impl Encodable for G1Commitment {
+    fn rlp_append(&self, s: &mut RlpStream) {
+        s.begin_list(2);
+        s.append(&self.x);
+        s.append(&self.y);
+    }
+}
+
+impl From<DisperserG1Commitment> for G1Commitment {
+    fn from(value: DisperserG1Commitment) -> Self {
+        Self {
+            x: value.x,
+            y: value.y,
+        }
+    }
+}
```

Comment:
Is there any point in redefining G1Commitment here? Its literally the same struct as the one in generated/common.rs. Perhaps we can just use that one directly?